docs/develop/python/message-passing.mdx

```diff
@@ -301,47 +301,112 @@ To send a Query to a Workflow Execution from Client code, use the `query()` meth
 
-### How to define an Update {#define-update}
 
-Workflow Updates handlers are methods in your Workflow Definition designed to handle updates.
-These updates can be triggered during the lifecycle of a Workflow Execution.
+### Writing Update handlers and validators as a Workflow author
 
-**Define an Update Handler**
+Here's a Workflow Definition that illustrates how to create an Update handler and validator:
 
-To define an update handler, use the [@workflow.update](https://python.temporal.io/temporalio.workflow.html#update) decorator on a method within your Workflow. This decorator can be applied to both asynchronous and synchronous methods.
+```python
+@dataclass
+class HelloWorldInput:
+    entity_to_be_greeted: str
 
-- **Decorator Usage:** Apply `@workflow.update` to the method intended to handle updates.
-- **Overriding:** If a method with this decorator is overridden, the overriding method should also be decorated with `@workflow.update`.
-- **Validator Method:** Optionally, you can define a validator method for the update handler. This validator is specified using `@update_handler_method_name.validator` and is invoked before the update handler.
-- **Method Parameters:** Update handlers should only use positional parameters. For non-dynamic methods, it's recommended to use a single parameter that is an object or data class, which allows for future expansion of fields.
-- **Return Values:** The update handler can return a serializable value. This value is sent back to the caller of the update.
 
-```python
-# ...
-    @workflow.update
-    async def update_workflow_status(self) -> str:
-        self.is_complete = True
-        return "Workflow status updated"
+@workflow.defn
+class HelloWorldWorkflow:
+    def __init__(self):
+        self.entity_to_be_greeted: Optional[str] = None
+
+    @workflow.run
+    async def run(self) -> str:
+        await workflow.wait_condition(lambda: self.entity_to_be_greeted is not None)
+        return self.greeting()
+
+    @workflow.updateap
+    def set_greeting(self, input: HelloWorldInput) -> str:
+        self.entity_to_be_greeted = input.entity_to_be_greeted
+        return self.greeting()
+
+    @set_greeting.validator
+    def set_greeting_validator(self, input: HelloWorldInput) -> None:
+        if input.entity_to_be_greeted not in {"world", "World"}:
+            raise Exception(f"invalid entity: {input.entity_to_be_greeted}")
+
+    def greeting(self) -> str:
+        return f"Hello, {self.entity_to_be_greeted}!"
 ```
 
-### Send an Update from a Temporal Client {#send-update-from-client}
+Note the following:
+
+- Update handlers and validators are defined using decorators, in a similar way to Signal and Query handlers.
+- This handler method does not do any long-running async operations; if it did, it would need to be an `async def`.
```

Comment:
Alternatively perhaps we want the handler in the example code to be `async`, although making that be necessary would complicate the code.